libraries/SoaringController/SoaringController.cpp

```diff
@@ -0,0 +1,415 @@
+#include "SoaringController.h"
+
+extern const AP_HAL::HAL& hal;
+
+
+// ArduSoar parameters
+const AP_Param::GroupInfo SoaringController::var_info[] = {
+    // @Param: SOAR_ACTIVE
+    // @DisplayName: Is the soaring mode active or not
+    // @Description: Toggles the soaring mode on and off
+    // @Units: boolean
+    // @Range: 0 1
+    // @User: Advanced
+    AP_GROUPINFO("ACTIVE", 0, SoaringController, soar_active, 0),
```

Comment:
Yes, thanks for the heads-up, I'm aware of this nastiness, after killing quite a bit of time over a similar issue before :)